gmso/core/atom.py

```diff
@@ -0,0 +1,137 @@
+
+import unyt as u
+
+from pydantic import Field, validator
+
+from gmso.core.element import Element
+from gmso.abc.abstract_site import Site
+from gmso.core.atom_type import AtomType
+from gmso.utils._constants import UNIT_WARNING_STRING
+
+
+class Atom(Site):
+    __base_doc__ = """An atom represents a single element association in a topology.
+    
+    Atoms are the representation of an element within `gmso` that describes any general 
+    atom in a molecular simulation. Atoms also contain information that are unique to 
+    elements vs other types of interaction sites in molecular simulations. 
+    For example, charge, mass, and periodic table information.
+
+    Notes
+    -----
+    Atoms have all the attributes inherited from the base Site class
+
+    Examples
+    --------
+        >>> from gmso.core.atom import Atom
+        >>> atom1 = Atom(name='lithium')
+
+    See Also
+    --------
+    gmso.abc.AbstractSite
+        An Abstract Base class for implementing site objects in GMSO. The class Atom bases from
+        the gmso.abc.abstract site class
+    """
+    charge_: Optional[Union[u.unyt_quantity, float]] = Field(
+        None,
+        description='Charge of the atom',
+    )
+
+    mass_: Optional[Union[u.unyt_quantity, float]] = Field(
+        None,
+        description='Mass of the atom'
+    )
+
+    element_: Optional[Element] = Field(
+        None,
+        description='Element associated with the atom'
+    )
+
+    atom_type_: Optional[AtomType] = Field(
+        None,
+        description='AtomType associated with the atom'
+    )
+
+    @property
+    def charge(self) -> Union[u.unyt_quantity, None]:
+        charge = self.__dict__.get('charge_', None)
+        atom_type = self.__dict__.get('atom_type_', None)
+        if charge is not None:
+            return charge
+        elif atom_type is not None:
+            return atom_type.charge
+        else:
+            return None
+
+    @property
+    def mass(self) -> Union[u.unyt_quantity, None]:
+        mass = self.__dict__.get('mass_', None)
+        atom_type = self.__dict__.get('atom_type_', None)
+        if mass is not None:
+            return mass
+        elif atom_type is not None:
+            return atom_type.mass
+        else:
+            return None
+
+    @property
+    def element(self) -> Union[Element, None]:
+        return self.__dict__.get('element_', None)
+
+    @property
+    def atom_type(self) -> Union[AtomType, None]:
+        return self.__dict__.get('atom_type_', None)
+
+    @validator('charge_')
+    def is_valid_charge(cls, charge):
+        if charge is None:
+            return None
+        if not isinstance(charge, u.unyt_array):
+            warnings.warn(UNIT_WARNING_STRING.format('Charges', 'elementary charge'))
+            charge *= u.elementary_charge
+
+        elif charge.units.dimensions != u.elementary_charge.units.dimensions:
+            warnings.warn(UNIT_WARNING_STRING.format('Charges', 'elementary charge'))
```

Comment:
This warning would not be the exact same case. The first case is when no unyt there, the second case is when the unyt "unit" does not simplify to a "charge" unit dimension.

Reply:
Second case here would be an error. Units are in a unit that cannot be coerced into charge. (length != charge)